### PR TITLE
OSDOCS-3663: Various changes to ROSA prereqs

### PR DESCRIPTION
**Pull request by EricPonvelle**
Version(s):
4.10+

Issue:
This PR adds changes that cover the following JIRA cards:

1. https://issues.redhat.com/browse/OSDOCS-3564
2. https://issues.redhat.com/browse/OSDOCS-3663
2. https://issues.redhat.com/browse/OSDOCS-3664
3. https://issues.redhat.com/browse/OSDOCS-3665
4. https://issues.redhat.com/browse/OSDOCS-3666
5. https://issues.redhat.com/browse/OSDOCS-3667
6. https://issues.redhat.com/browse/OSDOCS-3669

Link to docs preview:

- **_[OSDOCS-3564](https://issues.redhat.com/browse/OSDOCS-3564)_**: This section is ported from OCP - https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_planning/rosa-sts-aws-prereqs.html#installation-aws-route53_rosa-sts-aws-prereqs
- **_[OSDOCS-3663](https://issues.redhat.com/browse/OSDOCS-3663)_**: Removed items from the list -  https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_planning/rosa-sts-aws-prereqs.html#rosa-account_rosa-sts-aws-prereqs
- **_[OSDOCS-3664](https://issues.redhat.com/browse/OSDOCS-3664)_**: Added [a reference to check the additional resources for a link on ELBs](https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_planning/rosa-sts-aws-prereqs.html#rosa-account_rosa-sts-aws-prereqs). I also modified [the ELB destination a bit](https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_support/rosa-troubleshooting-deployments.html#rosa-troubleshooting-general-deployment-elb_rosa-troubleshooting-cluster-deployments) - https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_planning/rosa-sts-aws-prereqs.html#additional-resources
- **_[OSDOCS-3665](https://issues.redhat.com//browse/OSDOCS-3665)_** - Cleaned the language that these roles are attaching to your AWS account via the ARN: https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_planning/rosa-sts-aws-prereqs.html#rosa-associating-account_rosa-sts-aws-prereqs
- **_[OSDOCS-3666](https://issues.redhat.com/browse/OSDOCS-3666)_** - I added [a reference to the additional resources](https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_planning/rosa-sts-aws-prereqs.html#rosa-account_rosa-sts-aws-prereqs) to the documentation. The [link in additional resources links ](https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_planning/rosa-limits-scalability.html#rosa-limits-scalability) to the AWS limits: https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_planning/rosa-sts-aws-prereqs.html#additional-resources 
- **_[OSDOCS-3667](https://issues.redhat.com/browse/OSDOCS-3667)_** - I simply added some text to point to the "AWS firewall preqs" in the same topic: https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_planning/rosa-sts-aws-prereqs.html#rosa-security-requirements_rosa-sts-aws-prereqs
- **_[OSDOCS-3669](https://issues.redhat.com/browse/OSDOCS-3669)_** - I added the command to verify the quota to the main bulleted list: https://people.redhat.com/eponvell/053122/OSDOCS-3663_Prereq-Improvements/rosa_planning/rosa-sts-aws-prereqs.html#rosa-account_rosa-sts-aws-prereqs

Additional information:
This PR tracks changes from 7 JIRA cards. The changes are individually small, and I note which link relates to which JIRA card.

### CODE REVIEW

File: modules/rosa-sts-aws-requirements.adoc [hunk 1]
```diff
@@ -24,7 +22,7 @@ Customers are encouraged, but not mandated, to deploy resources in a Virtual Pri
 [id="rosa-associating-account_{context}"]
 == Associating your AWS account
 
-To perform {product-title} (ROSA) cluster provisioning tasks, you must create `ocm-role` and `user-role` IAM resources in your AWS account and link them to your Red Hat organization.
+{product-title} (ROSA) cluster provisioning tasks require linked `ocm-role` and `user-role` {cluster-manager} IAM resources to your AWS account using your Amazon Resource Name (ARN).
```
Comment (mburke5678) [inline modules/rosa-sts-aws-requirements.adoc:25]: s/cluster provisioning/cluster-provisioning ?? (I had to read it a few times to understand.)

s/linked/linking ?? require the `ocm-role` and `user-role` {cluster-manager} IAM resources be linked to   (Seems like there needs to be another verb somewhere.)

File: modules/rosa-sts-aws-requirements.adoc [hunk 2]
```diff
@@ -9,12 +9,10 @@ The following prerequisites must be complete before you deploy a {product-title}
 
 [id="rosa-account_{context}"]
 == Account
-* The customer ensures that the link:https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html[AWS limits] are sufficient to support {product-title} provisioned within the customer's AWS account.
+* The customer ensures that the AWS limits are sufficient to support {product-title} provisioned within the customer's AWS account. Running `rosa verify quota` in the CLI validates that you have the required quota to run a cluster. See Additional resources for more information.
```
Comment (arendej) [inline modules/rosa-sts-aws-requirements.adoc:12]: Can we also add
> Please note that quota verification takes into account your AWS quota, but does note compare your consumption versus your quota.
.. or similar

**Review by arendej — CHANGES_REQUESTED** [review]
see comment

File: modules/rosa-sts-aws-requirements.adoc [hunk 3]
```diff
@@ -9,22 +9,26 @@ The following prerequisites must be complete before you deploy a {product-title}
 
 [id="rosa-account_{context}"]
 == Account
-* The customer ensures that the link:https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html[AWS limits] are sufficient to support {product-title} provisioned within the customer's AWS account.
+* You must ensure that the AWS limits are sufficient to support {product-title} provisioned within the customer's AWS account. Running `rosa verify quota` in the CLI validates that you have the required quota to run a cluster.
++
+[NOTE]
+====
+Quota verification checks your AWS quota, but it does note compare your consumption versus your AWS quota. See the "Limits and scalability" link in the Additional resources for more information.
```
Comment (tmalove) [inline modules/rosa-sts-aws-requirements.adoc:16]:
```suggestion
Quota verification checks your AWS quota, but it does not compare your consumption to your AWS quota. See the "Limits and scalability" link in Additional resources for more information.
```

File: modules/rosa-sts-aws-requirements.adoc [hunk 4]
```diff
@@ -9,22 +9,26 @@ The following prerequisites must be complete before you deploy a {product-title}
...
+====
++
 * If SCP policies are applied and enforced, these policies must not be more restrictive than the roles and policies required by the cluster.
-* The customer's AWS account should not be transferable to Red Hat.
-* The customer should not impose additional AWS usage restrictions beyond the defined roles and policies on Red Hat activities. Imposing restrictions will severely hinder Red Hat's ability to respond to incidents.
-* The customer may deploy native AWS services within the same AWS account.
-* The account must have a service-linked role set up as it is required for elastic load balancers (ELBs) to be configured.
+* You may deploy native AWS services within the same AWS account.
+* Your account must have a service-linked role set up as it is required for elastic load balancers (ELBs) to be configured. See AWS Limits in the  Additional resources for information on creating a service-linked role for your ELB if you have not created a load balancer in your AWS account previously.
```
Comment (tmalove) [inline modules/rosa-sts-aws-requirements.adoc:21]:
```suggestion
* Your account must have a service-linked role set up as it is required for elastic load balancers (ELBs) to be configured. See AWS Limits in the  Additional resources for information about creating a service-linked role for your ELB if you have not created a load balancer in your AWS account previously.
```

File: modules/rosa-sts-aws-requirements.adoc [hunk 5]
```diff
@@ -124,8 +128,9 @@ If you do not specify a profile, the default AWS profile is used.
 == Access requirements
 
 * Red Hat must have AWS console access to the customer-provided AWS account. This access is protected and managed by Red Hat.
-* The customer must not utilize the AWS account to elevate their permissions within the {product-title} cluster.
+* You must not use the AWS account to elevate their permissions within the {product-title} cluster.
```
Comment (tmalove) [inline modules/rosa-sts-aws-requirements.adoc:131]:
```suggestion
* You must not use the AWS account to elevate your permissions within the {product-title} cluster.
```

File: modules/rosa-sts-aws-requirements.adoc [hunk 6]
```diff
@@ -124,8 +128,9 @@ If you do not specify a profile, the default AWS profile is used.
...
 * Actions available in the `rosa` CLI utility or {cluster-manager-url} console must not be directly performed in the customer's AWS account.
+* You do not need to have a preconfigured domain to deploy ROSA clusters. If you wish to use a custom domain, see the Additional resources for information.
```
Comment (tmalove) [inline modules/rosa-sts-aws-requirements.adoc:133]:
```suggestion
* You do not need to have a preconfigured domain to deploy ROSA clusters. If you want to use a custom domain, see Additional resources for information.
```

File: modules/rosa-sts-aws-requirements.adoc [hunk 7]
```diff
@@ -138,4 +143,4 @@ If you do not specify a profile, the default AWS profile is used.
 == Security requirements
 * Volume snapshots will remain within the customer's AWS account and customer-specified region.
 * Red Hat must have ingress access to EC2 hosts and the API server from allow-listed IP addresses.
-* Red Hat must have egress allowed to the documented domains.
+* Red Hat must have egress allowed to the documented domains. See the "AWS firewall prerequisites" section for the designated domains.
```
Comment (jeana-redhat) [inline modules/rosa-sts-aws-requirements.adoc:146]: If you split the module as suggested, you could even do `.Additional resources` links to sections farther down the page (I have had to do this for similar situations)

File: modules/rosa-troubleshooting-deployment.adoc [hunk 8]
```diff
@@ -41,7 +41,8 @@ $ rosa init --delete-stack
 $ rosa init
 ----
 
-== Elastic load balancer (ELB) `AccessDenied` error
+[id="rosa-troubleshooting-general-deployment-elb_{context}"]
+== Creating the service role for the Elastic load balancer (ELB)
```
Comment (jeana-redhat) [inline modules/rosa-troubleshooting-deployment.adoc:45]:
```suggestion
== Creating the service role for the elastic load balancer (ELB)
```

File: rosa_planning/rosa-sts-aws-prereqs.adoc [hunk 9]
```diff
@@ -34,3 +34,6 @@ include::modules/osd-aws-privatelink-firewall-prerequisites.adoc[leveloffset=+1]
 == Additional resources
```
Comment (jeana-redhat) [inline rosa_planning/rosa-sts-aws-prereqs.adoc:34]: Needs a unique ID, e.g. from guidelines:
```
[role="_additional-resources"]
[id="additional-resources_configuring-alert-notifications"]
== Additional resources
```
But! if you break the stuff I mentioned above into modules, it is possible all of these would be distributed throughout the assembly under the modules they go with anyway. So use your judgement here :slightly_smiling_face:

**Review by jeana-redhat — COMMENTED** [review]
A couple really small nits and one very big annoying suggestion (sorry about that one)

**Comment by EricPonvelle** [comment]
I'll address @jeana-redhat's module split comments in #46287.

**Pull request merged**